test/e2e/framework/pod/pod_client.go

```diff
@@ -89,43 +90,84 @@ func PodClientNS(f *framework.Framework, namespace string) *PodClient {
 type PodClient struct {
 	f *framework.Framework
 	v1core.PodInterface
-	namespace string
+	namespace  string
+	trackOwner bool
+}
+
+// WithOwnerTracking enable automatic add of annotations recording the code location
+// which created a pod. This is helpful when troubleshooting e2e tests (like e2e_node)
+// which leak pods because insufficient cleanup.
+func (c *PodClient) WithOwnerTracking() *PodClient {
+	c.trackOwner = true
+	return c
 }
 
 // Create creates a new pod according to the framework specifications (don't wait for it to start).
 func (c *PodClient) Create(ctx context.Context, pod *v1.Pod) *v1.Pod {
-	c.mungeSpec(pod)
-	p, err := c.PodInterface.Create(ctx, pod, metav1.CreateOptions{})
-	framework.ExpectNoError(err, "Error creating Pod")
-	return p
+	// we need to compute the call stack manually
+	// test-code                   # 3
+	// +- Create                   # 2
+	//    +- createWithOwner       # 1
+	//       +- setOwnerAnnotation # 0
+	//          +- runtime.Caller  # see https://pkg.go.dev/runtime#Caller
+	return c.createWithOwner(ctx, pod, 3)
```

Comment:
There's no guarantee that `Create` gets called directly by test code. It might get called by another helper.

A more reliable way to determine the caller is to invoke `ginkgo.GinkgoHelper` in all helper functions, like this one here.

Then call [NewCodeLocation](https://pkg.go.dev/github.com/onsi/ginkgo/v2/types#NewCodeLocation) with zero skip value to get the caller.

Reply:
TIL, thanks! Changing accordingly